src/util/simplify_expr.cpp

```diff
@@ -1680,15 +1680,19 @@ simplify_exprt::simplify_byte_extract(const byte_extract_exprt &expr)
     {
       return op_byte_update.value();
     }
-    else if(
-      el_size.has_value() &&
-      *el_size <= pointer_offset_bits(op_byte_update.value().type(), ns))
```

Comment:
Huh.. wasn't there even a compiler warning on that?

Reply:
This was invoking this piece of code:
https://github.com/diffblue/cbmc/blob/8848ad878da46bbcd0cedad634e235e6faf8cb17/src/nonstd/optional.hpp#L997..L1000
which will return `false` when the `optionalt` doesn't have a value.